Add snowflake plus source docs (#2450)
dlt-hub/dlt: docs/website/docs/plus/ecosystem/snowflake_plus.md

---
title: Snowflake Plus
description: Snowflake destination with Iceberg table support
keywords: [Snowflake, Iceberg, destination]
---

# Snowflake Plus

> Snowflake Plus is an **experimental** extension of the [snowflake destination](../../dlt-ecosystem/destinations/snowflake.md) that adds Apache Iceberg table creation and related features.

Reviewer: ```suggestion
Snowflake Plus is an **experimental** extension of the [Snowflake destination](../../dlt-ecosystem/destinations/snowflake.md) that adds Apache Iceberg table creation and related features.
```

> Snowflake Plus is an **experimental** extension of the [Snowflake destination](../../dlt-ecosystem/destinations/snowflake.md) that adds Apache Iceberg table creation and related features.

Reviewer: ```suggestion
Snowflake Plus is an **experimental** extension of the [Snowflake destination](../../dlt-ecosystem/destinations/snowflake.md) that adds [Apache Iceberg tables](https://docs.snowflake.com/en/user-guide/tables-iceberg) creation and related features.
```

> This destination is available starting from `dlt-plus==0.9.0` It fully supports all the functionality of the standard Snowflake destination, plus:

Reviewer: ```suggestion
This destination is available starting from `dlt-plus==0.9.0`. It fully supports all the functionality of the standard Snowflake destination, plus:
```

1. The ability to create Iceberg tables in Snowflake by setting `force_iceberg` to `true` in your `config.toml` file or `dlt.yml` file.
2. Additional configuration for Iceberg tables in Snowflake via:
   - `external_volume`: The external volume name where Iceberg data is stored.
   - `catalog`: The catalog name in which Iceberg tables are created. Defaults to `"SNOWFLAKE"`.
   - `base_location`: The base path that Snowflake uses for storing the table. If omitted, the table name is used.

## Installation

Install the `dlt-plus` package with the `snowflake` extra:

```sh
pip install "dlt-plus[snowflake]"
```

## Setup

> Once `snowflake` is installed, you can configure a pipeline to use `snowflake_plus` exactly the same way you would use `snowflake` or initialize a dlt+ project with the `snowflake_plus` source.

Reviewer: ```suggestion
Once `snowflake` extra is installed, you can configure a pipeline to use `snowflake_plus` exactly the same way you would use `snowflake` or initialize a dlt+ project with the `snowflake_plus` source.
```

### Use as a drop-in replacement for the `snowflake` destination

> 1. Make sure you have configured [Snowflake credentials](../../dlt-ecosystem/destinations/snowflake#setup-guide)

Reviewer: ```suggestion
1. Make sure you have configured [Snowflake credentials](../../dlt-ecosystem/destinations/snowflake.md#setup-guide)
```

> 2. Make sure you [set up database user and permissions](../../dlt-ecosystem/destinations/snowflake#set-up-the-database-user-and-permissions)

Reviewer: ```suggestion
2. Make sure you [set up database user and permissions](../../dlt-ecosystem/destinations/snowflake.md#set-up-the-database-user-and-permissions)
```

external_volume = "<external_volume_name>"
force_iceberg = true
```

6. Use the `snowflake_plus` destination in your pipeline:

```python
import dlt

pipeline = dlt.pipeline(
    pipeline_name="my_snowflake_plus_pipeline",
    destination="snowflake_plus",
    dataset_name="my_dataset"
)

@dlt.resource
def my_iceberg_table():
    ...
```

### Use with `dlt+` project

Initialize a dlt+ project in the current working directory. Replace `sql_database` with the source of your choice:

```sh
dlt project init sql_database snowflake_plus
```

This will create a Snowflake Plus destination in your `dlt.yml` file.

```yaml
destinations:
  snowflake_plus:
    type: snowflake_plus
```

To enable Iceberg table creation, set the `force_iceberg` option to `true` and `external_volume` to the name of the external volume you created in step 3.

```yaml
destinations:
  snowflake_plus:
    type: snowflake_plus
    force_iceberg: true
    external_volume: "<external_volume_name>"
```

## Configuration

The `snowflake_plus` destination extends the standard Snowflake configuration with additional options:

| Option | Description | Required | Default |
|--------|-------------|----------|---------|

> | `force_iceberg` | Whether to force the creation of Iceberg tables | No | False |

Reviewer: ```suggestion
| `force_iceberg` | Whether to force the creation of Iceberg tables | No | `False` |
```

| `external_volume` | The external volume to store Iceberg metadata | Yes | None |

> | `catalog` | The catalog to use for Iceberg tables | No | "SNOWFLAKE" |

Reviewer: ```suggestion
| `catalog` | The catalog to use for Iceberg tables | No | `"SNOWFLAKE"` |
```

> | `base_location` | Custom base location for Iceberg data in the external volume | No | Table name |

Reviewer: ```suggestion
| `base_location` | Custom base location for Iceberg data in the external volume | No | `<dataset_name>/<table_name>` |
```
